flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/stream/StreamingJoinOperator.java

```diff
@@ -289,27 +304,26 @@ protected boolean isHybridStreamBatchCapable() {
 
     @Override
     protected void emitStateAndSwitchToStreaming() throws Exception {
-        LOG.info("{} emit and switch to streaming", getPrintableName());
+        LOG.info("{} emit and switch to streaming leftIsOuter: {}, rightIsOuter: {}", getPrintableName(), leftIsOuter, rightIsOuter);
         if (leftIsOuter) {
             leftRecordStateView.emitCompleteState(getKeyedStateBackend(), this.collector,
-                    rightRecordStateView, joinCondition, false, true /* inputIsLeft */);
+                    rightRecordStateView, joinCondition, false, true, isEquijoin, true, statelessNullKeysEnabled, isBatchMode());
```

Comment:
you have isBatchMode plumbed through the APIs, but I didn't see actually get used (when we're emitting, we know that we're going from batch -> streaming, so when this is called, batchmode is always true

Reply:
Good point - addressed in latest commit